src/resources/js/v2/tickets-modal.js

```diff
@@ -0,0 +1,362 @@
+
+				if ( $cartItem.is( ':visible' ) ) {
+					const ticketID = $cartItem.closest( tribe.tickets.block.selectors.item ).data( 'ticket-id' );
+					const $ticketContainer = $( obj.selectors.metaForm ).find( '.tribe-tickets__item__attendee__fields__container[data-ticket-id="' + ticketID + '"]' );
+
+					// Ticket does not have meta - no need to jump through hoops ( and throw errors ).
+					if ( ! $ticketContainer.length ) {
+						return;
+					}
+
+					const $existing = $ticketContainer.find( obj.selectors.metaItem );
+					const qty = tribe.tickets.block.getQty( $cartItem );
+
+					if ( 0 >= qty ) {
+						$ticketContainer.removeClass( 'tribe-tickets--has-tickets' );
+						$ticketContainer.find( obj.selectors.metaItem ).remove();
+
+						return;
+					}
+
+					if ( $existing.length > qty ) {
+						const removeCount = $existing.length - qty;
+						$ticketContainer.find( '.tribe-ticket:nth-last-child( -n+' + removeCount + ' )' ).remove();
+
+					} else if ( $existing.length < qty ) {
+
+						const ticketTemplate = window.wp.template( 'tribe-registration--' + ticketID );
+						const counter = 0 < $existing.length ? $existing.length + 1 : 1;
+
+						$ticketContainer.addClass( 'tribe-tickets--has-tickets' );
+
+						for ( let i = counter; i <= qty; i++ ) {
+							const data = { attendee_id: i };
+
+							$ticketContainer.append( ticketTemplate( data ) );
+							obj.maybeHydrateAttendeeBlockFromLocal( $existing.length );
+						}
+					}
+				}
+			}
+		);
+
+		tribe.tickets.block.maybeShowNonMetaNotice( $form );
+		$document.trigger( 'tribe-ar-fields-appended' );
+		tribe.tickets.loader.hide( $form );
+	};
+
+	/**
+	 * Pre-fills the modal AR fields from supplied data.
+	 *
+	 * @since TBD
+	 *
+	 * @param {array} meta - Data to fill the form in with.
+	 * @param {number} length - Starting pointer for partial fill-ins.
+	 */
+	obj.preFillModalMetaForm = function( meta ) {
+		if ( undefined === meta || 0 >= meta.length ) {
+			return;
+		}
+
+		const $form = $( obj.selectors.metaForm );
+		const $containers = $form.find( '.tribe-tickets__item__attendee__fields__container' );
+
+		$.each( meta, function( idx, ticket ) {
+			let current = 0;
+			const $currentContainers = $containers.find( obj.selectors.metaItem ).filter( '[data-ticket-id="' + ticket.ticket_id + '"]' );
+
+			if ( ! $currentContainers.length ) {
+				return;
+			}
+
+			$.each( ticket.items, function( indx, data ) {
+				if ( 'object' !== typeof data ) {
+					return;
+				}
+
+				$.each( data, function( index, value ) {
+					const $field = $currentContainers.eq( current ).find( '[name*="' + index + '"]' );
+
+					if ( ! $field.is( ':radio' ) && ! $field.is( ':checkbox' ) ) {
+						$field.val( value );
+					} else {
+						$field.each( function() {
+							const $item = $( this );
+
+							if ( value === $item.val() ) {
+								$item.prop( 'checked', true );
+							}
+						} );
+					}
+				} );
+
+				current++;
+			} );
+		} );
```

Comment:
Separate methods.

Reply:
This is not ready for review, pushed so Scott could know where we were at.